### ISSUE TITLE
Failed to configure built-in redis server

### ISSUE THREAD

**Issue by ub333** [happypandax/happypandax#103]
**Describe the bug**
On load, HPX gives the following error:
`'Failed to configure built-in redis server, see logs/redis_stdout.log '
'Could not establish connection to redis server, aborting. '`

Application then exits.

**To Reproduce**
1. Launch HPX from terminal
2. See error

**Expected behavior**
Launch app successfully

**Logs**
redis_stdout.log is zero bytes

**Screenshots**
n/a

**Environment:**
 - OS: macOS 10.14.4, fresh install
 - Chrome 73
 - HPX Version 0.9.0

**Additional context**
Thought I might have arsed up my OS with a hodge-podge of ports and binaries and whatnot, so I reinstalled from scratch. Forgot to back up my HPX database so that's empty too. >_<

**Comment by twiddli** [comment]
Hi, can I see the contents in `logs/redis_stdout.log`

**Comment by twiddli** [comment]
Hi, could you try the new release and see if it still occurs?

**Comment by Kazakuri** [comment]
Yep, looks like it's still broken.

![https://i.imgur.com/j2pJdBG.jpg](https://i.imgur.com/j2pJdBG.jpg)

```
69126:M 28 Mar 18:29:04.590 * Increased maximum number of open files to 10032 (it was originally set to 256).
                _._
           _.-``__ ''-._
      _.-``    `.  `_.  ''-._           Redis 3.2.1 (8a2f3fb3/0) 64 bit
  .-`` .-```.  ```\/    _.,_ ''-._
 (    '      ,       .-`  | `,    )     Running in standalone mode
 |`-._`-...-` __...-.``-._|'` _.-'|     Port: 7009
 |    `-._   `._    /     _.-'    |     PID: 69126
  `-._    `-._  `-./  _.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |           http://redis.io
  `-._    `-._`-.__.-'_.-'    _.-'
 |`-._`-._    `-.__.-'    _.-'_.-'|
 |    `-._`-._        _.-'_.-'    |
  `-._    `-._`-.__.-'_.-'    _.-'
      `-._    `-.__.-'    _.-'
          `-._        _.-'
              `-.__.-'

69126:M 28 Mar 18:29:04.592 # Server started, Redis version 3.2.1
69126:M 28 Mar 18:29:04.593 * DB loaded from disk: 0.001 seconds
69126:M 28 Mar 18:29:04.593 * The server is now ready to accept connections on port 7009
69126:signal-handler (1553812144) Received SIGTERM scheduling shutdown...
69126:M 28 Mar 18:29:04.693 # User requested shutdown...
69126:M 28 Mar 18:29:04.693 * Saving the final RDB snapshot before exiting.
69126:M 28 Mar 18:29:04.694 * DB saved on disk
69126:M 28 Mar 18:29:04.694 # Redis is now ready to exit, bye bye...
```

**Comment by twiddli** [comment]
Could you provide your `error.log`?

**Comment by ub333** [comment]
Still getting same error with 0.9.1, but it's now writing to the redis_stdout.log file. Same output as shared by Kazakuri. 

error.log is zero bytes
[redis_stdout.log](https://github.com/happypandax/happypandax/files/3020577/redis_stdout.log)
[activity.log](https://github.com/happypandax/happypandax/files/3020578/activity.log)

**Comment by twiddli** [comment]
Try starting the redis server manually before starting hpx: ` ./bin/linux/redis-server --bind 127.0.0.1 --port 7009`
change `linux` to `osx` if on OS X

**Comment by Kazakuri** [comment]
I can try that out when I get home from work later today (~6 hours), it might be worth mentioning that I installed a local copy of redis last night and set happypandax to listen to that and it started up okay.

**Comment by Kazakuri** [comment]
Yep, I can confirm that starting `./bin/osx/redis-server --bind 127.0.0.1 --port 7009` manually before starting hpx lets it start up fine as well.

**Comment by ub333** [comment]
> Try starting the redis server manually before starting hpx: ` ./bin/linux/redis-server --bind 127.0.0.1 --port 7009`
> change `linux` to `osx` if on OS X

Yes, this works for me too.

**Comment by twiddli** [comment]
@Kazakuri was your `error.log` empty too?

**Comment by Kazakuri** [comment]
Yep, it has been empty this whole time.

**Comment by twiddli** [comment]
Can you try running `v0.9.2` with the `--debug` flag and let me see `debug.log`?

**Comment by Kazakuri** [comment]
[debug.log](https://github.com/happypandax/happypandax/files/3026081/debug.log)
[redis_stdout.log](https://github.com/happypandax/happypandax/files/3026082/redis_stdout.log)
[activity.log](https://github.com/happypandax/happypandax/files/3026083/activity.log)

**Comment by ub333** [comment]
Same here

[debug.log](https://github.com/happypandax/happypandax/files/3026128/debug.log)
[redis_stdout.log](https://github.com/happypandax/happypandax/files/3026129/redis_stdout.log)
[activity.log](https://github.com/happypandax/happypandax/files/3026130/activity.log)

**Comment by twiddli** [comment]
fixed in HPX `v0.10.0`
